### PR TITLE
I thought i got display working but realized my error while writing the pull request comment

### PR DESCRIPTION
**Pull request by nxnode**
i did pytest and i tested it in ipython and it's true that it IS passing the tests, BUT the assignment is:

> display() will return a unicode string representing the list as if it were a Python tuple literal: “(12, ‘sam’, 37, ‘tango’)”

and now that I have typed this and tried a few other things I realize my test is wrong. it should testing to assert ll.display() == "(3, "three", "nine")"

back to the drawing board. I know how to undo a merged pull request but not sure about undoing a pull request that hasn't been merged.

### CODE REVIEW

File: code_fellows/linked_list.py [hunk 1]
```diff
@@ -4,12 +4,21 @@
...
     def __init__(self, values=None):
         self.head = None
+        self._length = 0
         if values:
             for value in values:
                 self.head = Node(value, next=self.head)
+                self._length += 1
+
+    def __len__(self):
+        return self.size()
```
Comment (W-Ely) [inline code_fellows/linked_list.py]: `__len__` and `size` should both `return self._length`

File: tests/code_fellows/test_linked_list.py [hunk 2]
```diff
@@ -1,17 +1,20 @@
 # https://codefellows.github.io/sea-python-401d6/assignments/linked_list.html
 
 import pytest
-
+import sys
 from code_fellows.linked_list import LinkedList
+from io import StringIO
```
Comment (W-Ely) [inline tests/code_fellows/test_linked_list.py]: Nice find on the test for print!
these imports should be arranged per pep008 https://peps.python.org/pep-0008/#imports

> Imports should be grouped in the following order:
> 1. Standard library imports.
> 2. Related third party imports.
> 3. Local application/library specific imports.

> You should put a blank line between each group of imports.

so in this case:
```python
import sys
from io import StringIO

import pytest

from code_fellows.linked_list import LinkedList
```

**Review by W-Ely — CHANGES_REQUESTED** [review]

File: tests/code_fellows/test_linked_list.py (hunk 2 above)
Comment (W-Ely) [reply tests/code_fellows/test_linked_list.py]: might see if VS code has the plugin `isort` https://pypi.org/project/isort/

**Comment by nxnode** [comment]
There is a built in isort extension but I didn't have it set to automatically do it's thing. I changed it to sort on save.

My print test is failing because it's returning something like this:
`'<code_fellows.linked_list.LinkedList object at 0x7f438122d160>\n'`

So I removed all of the print stuff and will start a new branch for testing and merge this branch.

**Comment by W-Ely** [comment]
> So I removed all of the print stuff and will start a new branch for testing and merge this branch.

This exercise has followed an unexpected code/git flow.  The entirety of these kinds of exercises should be in a single reviewed and approved PR.  As if a new complete feature is being added.

So really instead of merging this, continuing to work on it until it is complete, and then merging it would be better.

If you want a new branch, a okay, branch off this branch, then merge in your changes to this branch once they are ready.

**Review by W-Ely — APPROVED** [review]

**Pull request merged**